pkg/deployer/baremetal/deployer_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package baremetal
+
+import (
+	"context"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/GreptimeTeam/gtctl/pkg/deployer/baremetal/component"
+	"github.com/GreptimeTeam/gtctl/pkg/deployer/baremetal/config"
+	"github.com/GreptimeTeam/gtctl/pkg/logger"
+)
+
+var L = logger.New(os.Stdout, 1, logger.WithColored())
+
+func TestNewDeployer(t *testing.T) {
+	homedir, err := os.UserHomeDir()
```

Comment:
suggestion: We should use a non-$HOME directory for unit testing. This can avoid polluting the user's directory. Maybe the current directory of test cases.

After testing finished, we should clean up the directory(maybe use `defer {}`).

Reply:
great idea!